### PR TITLE
Metricsdeployment

### PR DESCRIPTION
**Pull request by drewwells**

### CODE REVIEW

File: controllers/databaseclaim_controller.go [hunk 1]
```diff
@@ -293,9 +285,105 @@ func (r *DatabaseClaimReconciler) Reconcile(ctx context.Context, req ctrl.Reques
...
+		controllerutil.AddFinalizer(&dbClaim, dbFinalizerName)
+		if err := r.Update(ctx, &dbClaim); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	// FIXME: turn on metrics deployments later when testing on box-2 is available
+	// if err := r.createMetricsDeployment(ctx, dbClaim); err != nil {
+	// 	return ctrl.Result{}, err
+	// }
```
Comment (drewwells) [inline controllers/databaseclaim_controller.go:301]: Disabled until it we have a working box-2 with crossplane installed

File: Makefile [hunk 2]
```diff
@@ -160,15 +160,22 @@ docker-build-dbproxy:
 	cd dbproxy && docker build -t ${DBPROXY_IMG_PATH}:${TAG} .
 
 .PHONY: docker-build
-docker-build: #test ## Build docker image with the manager.
+docker-build: .image-${TAG} #test ## Build docker image with the manager.
+
+
+.image-${TAG}:
 	docker build -t ${IMG_PATH}:${TAG} .
+	@touch $@
```
Comment (kd7lxl) [inline Makefile:168]: Doesn't docker do this kind of caching internally? What benefit do we get from doing it explicitly?

File: Makefile [hunk 3]
```diff
@@ -327,18 +335,14 @@ clean:
...
+deploy: .id docker-push build-chart
+	helm upgrade --install `cat .id`-db-ctrl ${IMAGE_NAME}-${CHART_VERSION}.tgz \
+		--debug --wait --namespace "`cat .id`" \
 	 	--create-namespace \
-                 -f helm/db-controller/minikube.yaml \
+	 	-f helm/db-controller/minikube.yaml \
 	 	--set dbController.class=`cat .id` \
-		--set image.tag="${TAG}" \
+	 	--set image.tag="${TAG}" \
 	 	--set db.identifier.prefix=`cat .id` ${HELM_SETFLAGS}
```
Comment (kd7lxl) [inline Makefile:345]: Does this indentation work? Some lines have mixed tabs and spaces, others just tabs.

File: pkg/postgres-exporter/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,132 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Release }}-{{ .Name }}
+  namespace: {{ .Namespace }}
+  ownerReferences: {{ .DBClaimOwnerRef }}
+  labels:
+    app: {{ .Name }}
+    release: {{ .Release }}
```
Comment (kd7lxl) [inline pkg/postgres-exporter/deployment.yaml:9]: Recommended labels: https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels

**Review by kd7lxl — APPROVED** [review]

**Review by bjeevan-ib — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (drewwells) [reply Makefile:168]: saves a few seconds. docker seems to have its own way of declaring that it needs to rebuild an image. Im bypassing all of that with make

File: pkg/postgres-exporter/deployment.yaml (hunk 4 above)
Comment (drewwells) [reply pkg/postgres-exporter/deployment.yaml:9]: 🫠, not sure what I want to do with labels. perhaps inherit some from dbclaim too?

File: Jenkinsfile [hunk 5]
```diff
@@ -24,25 +24,13 @@ pipeline {
...
-      steps {
-      withCredentials([string(credentialsId: 'GITHUB_TOKEN', variable: 'GitHub_PAT')]) {
-          sh "echo machine github.com login $GitHub_PAT > ~/.netrc"
-          sh "echo machine api.github.com login $GitHub_PAT >> ~/.netrc"
-        }
-		dir("$DIRECTORY") {
-		  sh "sudo apt-get update"
-		  sh "sudo apt-get -y install postgresql-client"
-		  sh "echo 'db-controller-name' > .id"
-		  sh "make test"
```
Comment (kd7lxl) [inline Jenkinsfile:37]: classic way to pass the tests

**Review by kd7lxl — APPROVED** [review]

**Pull request closed without merging**